android/app/src/main/java/org/fossasia/openevent/core/track/session/SessionViewHolder.java

```diff
@@ -1,66 +1,282 @@
+    }
+
+    private void setSessionStatus() {
+        ZonedDateTime start = DateConverter.getDate(session.getStartsAt());
+        ZonedDateTime end = DateConverter.getDate((session.getEndsAt()));
+        ZonedDateTime current = ZonedDateTime.now();
+        if (DateService.isUpcomingSession(start, end, current)) {
+            sessionStatus.setVisibility(View.VISIBLE);
+            sessionStatus.setText(R.string.status_upcoming);
+        } else if (DateService.isOngoingSession(start, end, current)) {
+            sessionStatus.setVisibility(View.VISIBLE);
+            sessionStatus.setText(R.string.status_ongoing);
+        }
+
+    }
+
+    private void handleVisibilityByType(int type) {
+        switch (type) {
+            case trackWiseSessionList:
+                trackImageIcon.setVisibility(View.GONE);
+                sessionTrack.setVisibility(View.GONE);
+                break;
+
+            case locationWiseSessionList:
+                sessionLocation.setVisibility(View.GONE);
+                locationIcon.setVisibility(View.GONE);
+                break;
+
+            case speakerWiseSessionList:
+                sessionSpeaker.setVisibility(View.GONE);
+                speakerIcon.setVisibility(View.GONE);
+                break;
+
+            default: // Shouldn't reach here
+        }
+    }
+
+    private void setBookmarkClickListener(RealmDataRepository realmRepo, Track track) {
+
+        final int sessionId = session.getId();
+        sessionBookmarkIcon.setOnClickListener(v -> {
+            if (track == null) return;
+
+            if (session.getIsBookmarked()) {
+
+                realmRepo.setBookmark(sessionId, false).subscribe();
+                setBookmarkIcon(sessionBookmarkIcon, false, track.getFontColor());
+
+                if (onBookmarkSelectedListener != null)
+                    onBookmarkSelectedListener.showSnackbar(new BookmarkStatus(Color.parseColor(track.getColor()),
+                            sessionId, BookmarkStatus.Status.CODE_UNDO_REMOVED));
+
+            } else {
+                NotificationUtil.createNotification(session, context).subscribe(
+                        () -> {
+                            if (onBookmarkSelectedListener != null)
+                                onBookmarkSelectedListener.showSnackbar(new BookmarkStatus(Color.parseColor(track.getColor()),
+                                        sessionId, CODE_UNDO_ADDED));
+                        },
+                        throwable -> {
+                            if (onBookmarkSelectedListener != null)
+                                onBookmarkSelectedListener.showSnackbar(new BookmarkStatus(-1, -1, CODE_ERROR));
+                        });
+
+                realmRepo.setBookmark(sessionId, true).subscribe();
+                setBookmarkIcon(
+                        sessionBookmarkIcon, true, track.getFontColor());
+            }
+            WidgetUpdater.updateWidget(context);
+        });
+    }
```

Comment:
Add a todo to move all this unrelated to ViewHolder logic to somewhere else. A ViewHolder should only handle setting up UI